src/tools/pub/cmd/pub-add.md

```diff
@@ -0,0 +1,70 @@
+---
+title: dart pub add
+description: Use dart pub add to add a dependency.
+toc: false
+---
+
+_Add_ is one of the commands of the [pub tool](/tools/pub/cmd).
+
+```
+$ dart pub add <package>[:<constraint>] [options]
+```
+
+This command adds the specified package to the pubspec as a dependency,
+and then gets the dependency.
+
+For example, the following command is equivalent to
+editing `pubspec.yaml` (adding `http: ^0.12.1` under `dependencies`)
+and then calling `dart pub get`:
+
+```terminal
+$ dart pub add http:^0.12.1
+```
+
+By default, `dart pub add` gets the package from pub.dev.
+You can use options to specify other
+[dependency sources](/tools/pub/dependencies#dependency-sources) —
+Git repos, the Flutter SDK, alternate package servers, or
+a path within your local file system.
+
+
+## Options
+
+For options that apply to all pub commands, see
+[Global options](/tools/pub/cmd#global-options).
+
+<dl>
+    <dt><code>-d, --dev</code></dt>
+        <dd>Adds the package as a dev dependency,
+        instead of as a regular dependency.</dd>
```

Comment:
You might want to high an example like `dart pub add --dev test` which adds `package:test` as a dev-dependency.

Reply:
Okay I will add this

Reply:
@jp803333 I'll handle this too, since I'll be in there anyway.